aggregator/fetchers/factory.go

```diff
@@ -0,0 +1,71 @@
+package fetchers
+
+import "github.com/ElrondNetwork/elrond-sdk-erdgo/aggregator"
+
+// NewPriceFetcher returns a new price fetcher of the type provided
+func NewPriceFetcher(fetcherName string, responseGetter aggregator.ResponseGetter) (aggregator.PriceFetcher, error) {
+	if !isValidFetcherName(fetcherName) {
+		return nil, errInvalidFetcherName
+	}
+	if responseGetter == nil {
+		return nil, errNilResponseGetter
+	}
+
+	return createFetcher(fetcherName, responseGetter), nil
+}
+
+func isValidFetcherName(name string) bool {
```

Comment:
using this function and the knownFetchers var here is a little bit useless as you already do the switch anyway :)